build-tools/automation/yaml-templates/stage-msbuild-tests.yaml

```diff
@@ -0,0 +1,26 @@
+# Runs MSBuild tests that do not require devices on Mac/Windows
+
+parameters:
+  stageCondition: succeeded()
+
+stages:
+- stage: msbuild_dotnet
+  displayName: MSBuild Tests
+  dependsOn: mac_build
+  condition: ${{ parameters.stageCondition }}
+  jobs:
+  - template: run-msbuild-tests.yaml
+    parameters:
+      testOS: macOS
+      jobName: mac_msbuild_tests
+      jobDisplayName: macOS > Tests > MSBuild
+      agentCount: 8
+      testFilter: cat != SmokeTests
+      
+  - template: run-msbuild-tests.yaml
+    parameters:
+      testOS: Windows
+      jobName: win_msbuild_tests
+      jobDisplayName: Windows > Tests > MSBuild
+      agentCount: 4
+      testFilter: cat != SmokeTests
```

Comment:
I think we could move this block back into `azure-pipelines.yaml` to help cut down on the number of templates we use that aren't referenced more than once.

Reply:
My thought was that `azure-pipelines.yaml` is way too big and it takes too long to find the relevant stage yaml.  If each stage was its own template file we could just go directly to the stage we want.

But that's a stylistic opinion and I'm not committed to it if others feel different.  :grin:

Reply:
I wish yaml had some support around "go to reference" or "go to implementation", I'm often annoyed when working with multiple layers of nested templates having to jump back and forth between multiple page tabs rather than scrolling. That's also just my opinion though and am happy to go either way on this, maybe I can find some VS code extension to make my life easier...